python_frank_energie/frank_energie.py

```
@@ -1,646 +1,1434 @@
+            }
+            """,
+            "MonthSummary",
+            {"siteReference": site_reference},
+        )
+
+        try:
+            response = await self._query(query)
+            return MonthSummary.from_dict(response)
+        except Exception as e:
+            raise FrankEnergieException(
+                f"Failed to fetch month summary: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)